Main.py

```diff
@@ -43,27 +47,31 @@
             first_int_iter = 1
             record_count = full_page_record_count
             while record_count >= full_page_record_count:
-                riot_apida.call_api(curr_url)
-                record_count = len(riot_apida.data_payload)
-                riot_apida.data_payload[riot_apida.int_iter_keys[0]] = [first_int_iter] * record_count
-                curr_dashboard_row[riot_apida.int_iter_keys[0]] = first_int_iter
-                curr_dashboard_row['full_url'] = curr_url
+                status_code = riot_apida.call_api(curr_url)
+                curr_dashboard_row.loc[:, riot_apida.int_iter_keys[0]] = first_int_iter
+                curr_dashboard_row.loc[:,'full_url'] = curr_url
+                curr_dashboard_row.loc[:,'status_code'] = status_code 
                 current_dashboard = pd.concat([current_dashboard, curr_dashboard_row])
                 riot_apida.csv_upsave(curr_dashboard_row,
                                       "temp_dashboard"
                                       + str(int(curr_date.strftime("%Y%m%d%H%M%S")))
                                       + ".csv")
-                riot_apida.csv_upsave(riot_apida.data_payload,
-                                      "temp_data_payload" 
-                                      + str(int(curr_date.strftime("%Y%m%d%H%M%S"))) 
-                                      + ".csv")
+                if status_code == 200:
+                    record_count = len(riot_apida.data_payload)
```

Comment:
Do you do anything in the case that API is down (500 error, etc) to prevent an infinite loop? It seems like you'd need to set `record_count = 0` in the case of API error to prevent `while record_count >= full_page_record_count:` from going forever

Reply:
Ono, major duh moment. Thanks for the catch! Just made the change in the "chunkify" branch.